### PR TITLE
julia 1.10.3

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Also refines/updates the CPU JIT targets.

### CODE REVIEW

**Comment by fxcoudert** [comment]
I don't think we should be tweaking `JULIA_CPU_TARGET` ourselves. I've asked upstream what they use: https://github.com/JuliaLang/julia/issues/54319 for official builds

**Comment by fxcoudert** [comment]
Let's use upstream values everywhere, except Apple Silicon where we drop non-Apple hardware.

File: Formula/j/julia.rb [hunk 1]
```diff
@@ -91,25 +91,23 @@ def install
...
     cpu_targets = %w[generic]
     if Hardware::CPU.arm?
-      # Cortex A57, Thunder-X2, Nvidia Carmel, Neoverse V1/V2,
-      # Cortex A510/A710/A715/X2/X3
-      if OS.linux?
-        cpu_targets += %w[cortex-a57 thunderx2t99 carmel,clone_all neoverse-512tvb,base(3)
-                          armv8.5-a,sve2,sve2-bitperm,i8mm,bf16,base(3)]
+      if OS.mac?
+        # For Apple Silicon, we don't care about other hardware
+        cpu_targets << "apple-m1,clone_all"
```
Comment (carlocab) [inline Formula/j/julia.rb:101]:
```suggestion
        cpu_targets << "apple-m1,base(3)"
```
https://github.com/JuliaCI/julia-buildkite/blob/ea50eb719242bc3e844227a2ebf54a49d308bafc/utilities/build_envs.sh#L67

Comment (fxcoudert) [reply Formula/j/julia.rb:101]: `base(3)` means a profile is built on top of another base, and we don't want that here. We really want `apple-m1,clone_all`. Confirmed by upstream at https://github.com/JuliaLang/julia/issues/54319#issuecomment-2087587703

**Review by carlocab — APPROVED** [review]

**Pull request merged**